client/src/api/datasource/api-client.ts

```diff
@@ -2,20 +2,66 @@ import axios from 'axios';
 import { DataSourceClient } from './datasource-client';
 import { DataSource } from '@/api/Models';
 import { TraceabilityOrigin } from '@/utils/sigmfMetadata';
+import { AccountInfo, IPublicClientApplication } from '@azure/msal-browser';
 
 export class ApiClient implements DataSourceClient {
+  private instance: IPublicClientApplication;
+  private account: AccountInfo;
+
+  constructor(instance: IPublicClientApplication, account: AccountInfo) {
+    this.instance = instance;
+    this.account = account;
+  }
+
+  private async getAccessToken() {
+    const api_scope = 'api://' + import.meta.env.IQENGINE_APP_ID + '/api';
+    try {
+      const response = await this.instance.acquireTokenSilent({
+        account: this.account,
+        scopes: [api_scope],
+      });
+      return response.accessToken;
+    } catch (error) {
+      return null;
+    }
+  }
+
+  private async requestWithAuth(config) {
```

Comment:
Not sure this accurately describes the intention, maybe something like 'requestWithRequiredHeaders' or similar

Reply:
Disagree. The headers are generally provided. The procedure specifically adds the authorization header making a call with auth.

Reply:
But it doesn't always request with auth?

Reply:
If no bearer is provided in config it does not

Reply:
That's what I mean by the name is confusing